Source/WTF/wtf/Compiler.h

```diff
@@ -562,6 +562,16 @@
 #define IGNORE_NULL_CHECK_WARNINGS_BEGIN IGNORE_WARNINGS_BEGIN("nonnull")
 #define IGNORE_NULL_CHECK_WARNINGS_END IGNORE_WARNINGS_END
 
+/* NOESCAPE */
```

Comment:
Since the concept will be new to some people, maybe we should add a comment:

`This attribute promises that a value will only be used for the duration of the current function, and not stored to the heap or in a global for later use. You can apply it to local variables and function arguments.`

Reply:
Sure.

Reply:
I don't think we can apply it to a local variable though.

Reply:
Added:
```
/* This attribute promises that a function argumemnt will only be used for the duration of the function,
   and not stored to the heap or in a global state for later use. The compiler does not verify this claim. */
```